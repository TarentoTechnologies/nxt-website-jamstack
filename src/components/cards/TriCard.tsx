import { useState } from "react";
import React from "react";

import {
  bgDarkViridianGreen,
  bgLeitchsBlue,
  bgOxideBlue,
  bgRoslynBlue,
  bgSpectrumViolet,
  bgViridianGreen,
} from "../../styles/style-guide/ColorGuide.module.css";
import { PrimaryTitle } from "../titles/PrimaryTitle";
import { imgStyles, titleStyles, triCardStyles } from "./TriCard.module.css";

interface TriCardProps {
  title: string;
  logo: any;
  bgColor: any;
  clickHandler?: (event: any) => void;
}

export const TriCard = ({
  title,
  logo,
  bgColor,
  clickHandler,
}: TriCardProps) => {
  const bgBlue = bgColor === "blue" ? bgOxideBlue : "";
  const bgGreen = bgColor === "green" ? bgViridianGreen : "";
  const bgViolet = bgColor === "violet" ? bgSpectrumViolet : "";

  const [hovered, setHovered] = useState(false);

  const handleMouseEnter = () => {
    setHovered(true);
  };

  const handleMouseLeave = () => {
    setHovered(false);
  };

  const getHoverBgColor = () => {
    switch (bgColor) {
      case "blue":
        return bgLeitchsBlue;
      case "green":
        return bgDarkViridianGreen;
      case "violet":
        return bgRoslynBlue;
      default:
        return "";
    }
  };

  const hoverBgColor = hovered ? getHoverBgColor() : "";

  return (
    <div
      className={`container-fluid ${triCardStyles} ${bgBlue} ${bgGreen} ${bgViolet} ${hoverBgColor}`}
      onClick={clickHandler}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      data-testid="tri-card"
    >
      <div className={`${imgStyles}`}>
        <img
          src={logo}
          alt={title}
          title={title}
          data-aos="fade-up"
          data-aos-delay="500"
        />
      </div>
      <div className={`${titleStyles}`} data-aos="fade-up">
        <PrimaryTitle title={title} variant="card" color="white" />
      </div>
    </div>
  );
};
