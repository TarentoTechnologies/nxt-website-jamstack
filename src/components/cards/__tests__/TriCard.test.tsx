import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { TriCard } from "../TriCard";

describe("Test if the TriCard component is working as expected", () => {
  it("<TriCard /> matches snapshot", () => {
    const component = render(
      <TriCard
        title={"TECHNOLOGY"}
        logo={
          "https://strapi-stage-nxt.tarento.com/uploads/technology_de1bb99b6d.svg"
        }
        bgColor={"blue"}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<TriCard /> renders without exploding", () => {
    const component = render(
      <TriCard
        title={"TECHNOLOGY"}
        logo={
          "https://strapi-stage-nxt.tarento.com/uploads/technology_de1bb99b6d.svg"
        }
        bgColor={"blue"}
      />
    );
    expect(component.getByTestId("tri-card")).toBeInTheDocument();
  });
});
