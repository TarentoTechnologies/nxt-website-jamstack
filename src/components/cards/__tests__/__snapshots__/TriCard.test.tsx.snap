// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the TriCard component is working as expected <TriCard /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid triCardStyles bgOxideBlue   "
    data-testid="tri-card"
  >
    <div
      class="imgStyles"
    >
      <img
        alt="TECHNOLOGY"
        data-aos="fade-up"
        data-aos-delay="500"
        src="https://strapi-stage-nxt.tarento.com/uploads/technology_de1bb99b6d.svg"
        title="TECHNOLOGY"
      />
    </div>
    <div
      class="titleStyles"
      data-aos="fade-up"
    >
      <div
        class="
      undefined
        
        
        
        
        cardHeader
        
        
        
        
        
        
        white"
        data-aos="fade-up"
        data-testid="primary-title"
      >
        TECHNOLOGY
      </div>
    </div>
  </div>
</div>
`;
